crates/krun-sys/build.rs

```diff
@@ -1,18 +1,29 @@
 use std::env;
 use std::path::PathBuf;
 
-fn main() {
+fn main() -> Result<(), pkg_config::Error> {
     println!("cargo::rerun-if-changed=wrapper.h");
-    println!("cargo::rustc-link-lib=krun");
```

Comment:
I'm most likely asking out of ignorance here, but https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib says this emits the `-l` flag to the compiler.

With this gone, where / how does that happen then? Below we only emit `-I` flags, right?

Reply:
bindgen doesn't really need to link against the library, it just needs the headers.

Reply:
Uhh... But since this crate is `krun-sys` (and specifically https://doc.rust-lang.org/cargo/reference/manifest.html#the-links-field), I think it's expected that it'll actually link to libkrun...

Reply:
But also, I'm confused where the linking to libkrun happens if not here?

Reply:
I was also getting very confused by this, because I was seeing the library getting injected at build time. Turns out `pkg_config::probe_library("libkrun")?;` also injects (by printing to stdout) parameters to the linker. The documentation of `pkg-config` is sparse, at best.